xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp

```diff
@@ -1113,21 +1113,33 @@ void CWebVTTHandler::ConvertAddSubtitle(std::vector<subtitleData>* subList)
   // Convert tags and apply the CSS Styles converted
   ConvertSubtitle(m_subtitleData.text);
 
-  if (!subList->empty())
+  if (m_lastSubtitleData)
   {
-    // Youtube WebVTT can have multiple cues with same time, text and position
-    // sometimes with different css color but only last cue will be visible
-    // this cause unexpected results on screen so we keep only the last one
-    const subtitleData& prevSub = subList->back();
-    if (prevSub.startTime == m_subtitleData.startTime &&
-        prevSub.stopTime == m_subtitleData.stopTime && prevSub.textRaw == m_subtitleData.textRaw &&
-        prevSub.cueSettings == m_subtitleData.cueSettings)
+    // Check for same subtitle data
+    if (m_lastSubtitleData->startTime == m_subtitleData.startTime &&
+        m_lastSubtitleData->stopTime == m_subtitleData.stopTime &&
+        m_lastSubtitleData->textRaw == m_subtitleData.textRaw &&
+        m_lastSubtitleData->cueSettings == m_subtitleData.cueSettings)
     {
-      subList->pop_back();
+      if (subList->empty())
+      {
+        // On segmented WebVTT, can happens there are malformed streams that repeat the same
```

Comment:
```suggestion
        // On segmented WebVTT, it can happen that the same subtitle entry is sent on consecutive demux packets. Hence we avoid showing overlapping subs.
```